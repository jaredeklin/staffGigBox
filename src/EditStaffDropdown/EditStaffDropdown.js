import React, { Component } from 'react';
import PropTypes from 'prop-types';

export class EditStaffDropdown extends Component {
  constructor(props) {
    super(props);
    this.api = new Api();
    this.state = {
      staff_id: null,
      availableStaff: []
    };
  }

  handleChange = async event => {
    this.setState({
      staff_id: event.target.value
    });
  };

    this.props.updateEventStaff(this.state);
  };

  displayStaff = () => {
    return this.props.staff.map(person => {
      return (
        <option key={person.id} value={person.id}>
          {person.name}
        </option>
      );
    });
  };

  render() {
    return (
      <form>
        <select onChange={this.handleChange}>
          <option>Select staff</option>
          {this.displayStaff()}
        </select>
      </form>
    );
  }
}

EditStaffDropdown.propTypes = {
  updateEventStaff: PropTypes.func,
  event_id: PropTypes.number,
  staff: PropTypes.array
};
