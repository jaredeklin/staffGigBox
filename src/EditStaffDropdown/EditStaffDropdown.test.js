import React from 'react';
import { shallow } from 'enzyme';
import { EditStaffDropdown } from './EditStaffDropdown';

describe('EditStaffDropdown', () => {
  let wrapper;
  let mockStaff = [{ id: 2, name: 'Jared' }];
  let mockEvent = { event_id: 1, date: '2018-06-06' };
  let mockUpdateEventStaff = jest.fn();
  const mockCurrentPerson = { name: 'Jared', role: 'Bartender' };
  const mockCloseModal = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <EditStaffDropdown
        staff={mockStaff}
        event={mockEvent}
        updateEventStaff={mockUpdateEventStaff}
        currentPerson={mockCurrentPerson}
        closeModal={mockCloseModal}
      />,
      { disableLifecycleMethods: true }
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update state and call updateEventStaff', () => {
    const mockEvent = { target: { value: 'oh yeah' } };

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('staff_id')).toEqual('oh yeah');
  });
});
