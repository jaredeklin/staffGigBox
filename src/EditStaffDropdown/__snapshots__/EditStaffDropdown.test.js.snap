// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditStaffDropdown should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditStaffDropdown
    closeModal={[Function]}
    currentPerson={
        Object {
            "name": "Jared",
            "role": "Bartender",
          }
    }
    event={
        Object {
            "date": "2018-06-06",
            "event_id": 1,
          }
    }
    staff={
        Array [
            Object {
              "id": 2,
              "name": "Jared",
            },
          ]
    }
    updateEventStaff={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="modal-main"
>
        Replace 
        Jared
         in a role of 
        Bartender
         with
        <select
                onChange={[Function]}
        >
                <option>
                        Select staff
                </option>
        </select>
        <button
                onClick={[Function]}
        >
                Save
        </button>
        <button
                onClick={[Function]}
        >
                Close
        </button>
</section>,
      "className": "modal-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Replace ",
          "Jared",
          " in a role of ",
          "Bartender",
          " with",
          <select
            onChange={[Function]}
>
            <option>
                        Select staff
            </option>
</select>,
          <button
            onClick={[Function]}
>
            Save
</button>,
          <button
            onClick={[Function]}
>
            Close
</button>,
        ],
        "className": "modal-main",
      },
      "ref": null,
      "rendered": Array [
        "Replace ",
        "Jared",
        " in a role of ",
        "Bartender",
        " with",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option>
                Select staff
</option>,
              Array [],
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select staff",
              },
              "ref": null,
              "rendered": "Select staff",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Close",
          "type": "button",
        },
      ],
      "type": "section",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="modal-main"
>
          Replace 
          Jared
           in a role of 
          Bartender
           with
          <select
                    onChange={[Function]}
          >
                    <option>
                              Select staff
                    </option>
          </select>
          <button
                    onClick={[Function]}
          >
                    Save
          </button>
          <button
                    onClick={[Function]}
          >
                    Close
          </button>
</section>,
        "className": "modal-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Replace ",
            "Jared",
            " in a role of ",
            "Bartender",
            " with",
            <select
              onChange={[Function]}
>
              <option>
                            Select staff
              </option>
</select>,
            <button
              onClick={[Function]}
>
              Save
</button>,
            <button
              onClick={[Function]}
>
              Close
</button>,
          ],
          "className": "modal-main",
        },
        "ref": null,
        "rendered": Array [
          "Replace ",
          "Jared",
          " in a role of ",
          "Bartender",
          " with",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option>
                  Select staff
</option>,
                Array [],
              ],
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Select staff",
                },
                "ref": null,
                "rendered": "Select staff",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Close",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Close",
            "type": "button",
          },
        ],
        "type": "section",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
